### PR TITLE
Hide ToC during search if requested

### LINKED ISSUE #194812: ToC shows up while doing search while `workbench.settings.settingsSearchTocBehavior` set to `hide`
1. On any OS on the latest Stable or Insiders, navigate to the Settings editor
2. Set `workbench.settings.settingsSearchTocBehavior` to `hide`
3. Clear the search query
4. Search for a setting
5. :bug: The ToC still shows up

This issue started occurring after the Settings editor switched to using a split view to allow for a resizable table of contents. For reference, in v1.63.2, searching for a setting while `workbench.settings.settingsSearchTocBehavior` was set to `hide` would hide the ToC, and clearing out the search query would show the ToC again.

### PR DESCRIPTION
**Pull request by rzhao271**
Fixes #194812

Please see the issue for the actual and expected behaviours.

### CODE REVIEW

**Comment by rzhao271** [comment]
Will have to debug the smoke test first before this PR can be merged.

File: src/vs/workbench/contrib/preferences/common/preferencesContribution.ts [hunk 1]
```diff
@@ -155,7 +155,7 @@ registry.registerConfiguration({
 				nls.localize('settingsSearchTocBehavior.hide', "Hide the Table of Contents while searching. The search results will not be grouped by category, and instead will be sorted by similarity to the query, with exact keyword matches coming first."),
 				nls.localize('settingsSearchTocBehavior.filter', "Filter the Table of Contents to just categories that have matching settings. Clicking a category will filter the results to that category. The search results will be grouped by category."),
 			],
-			'description': nls.localize('settingsSearchTocBehavior', "Controls the behavior of the settings editor Table of Contents while searching."),
+			'description': nls.localize('settingsSearchTocBehavior', "Controls the behavior of the Settings editor Table of Contents while searching. If this setting is being changed in the Settings editor, the setting will take effect after the search query is modified."),
```
Comment (roblourens) [inline src/vs/workbench/contrib/preferences/common/preferencesContribution.ts:158]: I think it should either be "Settings Editor" or "settings editor"?

File: src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts [hunk 2]
```diff
@@ -1538,7 +1549,9 @@ export class SettingsEditor2 extends EditorPane {
 
 		if (query && query !== '@') {
 			query = this.parseSettingFromJSON(query) || query;
-			return this.triggerFilterPreferences(query);
+			return this.triggerFilterPreferences(query).then(() => {
```
Comment (roblourens) [inline src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts:1552]: You might wanna convert these methods to async/await as you go

**Review by roblourens — DISMISSED** [review]
Good find, sorry for leaving a mess

File: src/vs/workbench/contrib/preferences/common/preferencesContribution.ts (hunk 1 above)
Comment (rzhao271) [reply src/vs/workbench/contrib/preferences/common/preferencesContribution.ts:158]: I'm using the same casing as the release notes, where I believe it's always been "Settings editor"
Adding @gregvanl to the thread to double-check

Comment (gregvanl) [reply src/vs/workbench/contrib/preferences/common/preferencesContribution.ts:158]: Yep, we use "Settings editor" in documentation and release notes

File: src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts [hunk 3]
```diff
@@ -1518,7 +1512,24 @@ export class SettingsEditor2 extends EditorPane {
...
+	private toggleTocBySearchBehaviorType() {
+		const tocBehavior = this.configurationService.getValue<'filter' | 'hide'>(SEARCH_TOC_BEHAVIOR_KEY);
+		const hideToc = tocBehavior === 'hide';
+		if (hideToc) {
+			this.splitView.setViewVisible(0, false);
+			this.splitView.style({
+				separatorBorder: Color.transparent
+			});
+		} else {
+			this.splitView.setViewVisible(0, true);
```
Comment (rzhao271) [inline src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts:1528]: TODO: Check that the separator border colour is proper in this case

**Review by roblourens — APPROVED** [review]

**Pull request merged**